### PR TITLE
[CHK-1423] feat: update wallet api

### PR DESCRIPTION
**Pull request by pietro-tota**
### List of changes
Update wallet api

### Motivation and context
Update wallet api

### Type of changes

- [ ] Add new resources
- [ ] Update configuration to existing resources
- [ ] Remove existing resources

### Does this introduce a change to production resources with possible user impact?

- [ ] Yes, users may be impacted applying this change
- [x] No

### Does this introduce an unwanted change on infrastructure? Check terraform plan execution result

- [ ] Yes
- [x] No

### Other information

---

### CODE REVIEW

**Review by giovanniberti — DISMISSED** [review]

File: src/domains/wallet-app/api/wallet-service/v1/_openapi.json.tpl [hunk 1]
```diff
@@ -1,248 +1,368 @@
...
+          },
+          "paymentInstrumentId": {
+            "description": "Payment instrument identifier",
+            "type": "string"
+          },
+          "contractNumber": {
+            "description": "User contract identifier to be used with payment instrument to make a new payment",
+            "type": "string"
+          },
+          "gatewaySecurityToken": {
```
Comment (infantesimone) [inline src/domains/wallet-app/api/wallet-service/v1/_openapi.json.tpl:290]: better not to return this info to the client 😁

File: src/domains/wallet-app/api/wallet-service/v1/_openapi.json.tpl [hunk 2]
```diff
@@ -1,248 +1,368 @@
...
+            "description": "Wallet creation date",
+            "type": "string",
+            "format": "date-time"
+          },
+          "updateDate": {
+            "description": "Wallet update date",
+            "type": "string",
+            "format": "date-time"
+          },
+          "paymentInstrumentType": {
```
Comment (infantesimone) [inline src/domains/wallet-app/api/wallet-service/v1/_openapi.json.tpl:279]: maybe `walletType`, what do you think?

**Review by infantesimone — CHANGES_REQUESTED** [review]

**Comment by infantesimone** [comment]
cc @MarcoTormen

File: src/domains/wallet-app/api/wallet-service/v1/_openapi.json.tpl (hunk 1 above)
Comment (pietro-tota) [reply src/domains/wallet-app/api/wallet-service/v1/_openapi.json.tpl]: fixed with https://github.com/pagopa/pagopa-infra/pull/903/commits/d16239dffa5356b30293b97b85c34b6b83d69df7

File: src/domains/wallet-app/api/wallet-service/v1/_openapi.json.tpl [hunk 3]
```diff
@@ -1,248 +1,378 @@
...
+          "INITIALIZED",
+          "CREATED",
+          "ERROR"
+        ]
+      },
+      "CardPaymentInstrumentDetails": {
+        "type": "object",
+        "description": "Card payment instrument details",
+        "properties": {
+          "walletType": {
```
Comment (infantesimone) [inline src/domains/wallet-app/api/wallet-service/v1/_openapi.json.tpl:185]: duplicate? related to the parent?

Comment (pietro-tota) [reply src/domains/wallet-app/api/wallet-service/v1/_openapi.json.tpl:185]: This property needs to be here in order to be used as discriminator

File: src/domains/wallet-app/api/wallet-service/v1/_openapi.json.tpl [hunk 4]
```diff
@@ -1,248 +1,378 @@
...
+      "WalletStatus": {
+        "type": "string",
+        "description": "Enumeration of wallet statuses",
+        "enum": [
+          "INITIALIZED",
+          "CREATED",
+          "ERROR"
+        ]
+      },
+      "CardPaymentInstrumentDetails": {
```
[the remark this answers is not in the document]
Comment (pietro-tota) [reply src/domains/wallet-app/api/wallet-service/v1/_openapi.json.tpl:181]: I put here a "oneOf" with those details specific to card details that can be extended with other payment instrument types in feature. Maybe `CardWalletDetails`? What do you think? I made assumption about the fact that in feature can be other payment instrument that will be added to wallet other than cards

File: src/domains/wallet-app/api/wallet-service/v1/_openapi.json.tpl (hunk 3 above)
Comment (pietro-tota) [reply src/domains/wallet-app/api/wallet-service/v1/_openapi.json.tpl:185]: Fixed with https://github.com/pagopa/pagopa-infra/pull/903/commits/56916a1fbeeb4d69c23dca5331a4af6daa3cc855

**Review by infantesimone — APPROVED** [review]

**Pull request merged**